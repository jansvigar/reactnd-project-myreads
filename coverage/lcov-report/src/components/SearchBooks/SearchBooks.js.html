<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/SearchBooks/SearchBooks.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/SearchBooks</a> SearchBooks.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">9.09% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">14.29% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import * as BooksAPI from '../../BooksAPI';
import BooksGrid from '../BooksGrid/BooksGrid';
import SearchBox from './SearchBox';
import './SearchBooks.css';
&nbsp;
class SearchBooks extends Component {
  state = {
    books: []
  };
&nbsp;
  handleSearch = <span class="fstat-no" title="function not covered" >(q</span>uery) =&gt; {
<span class="cstat-no" title="statement not covered" >    BooksAPI.search(query).then(<span class="fstat-no" title="function not covered" >bo</span>oks =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      if(!books.error) {</span>
        let uniqueBooks = <span class="cstat-no" title="statement not covered" >removeDuplicates(books, 'id');</span>
<span class="cstat-no" title="statement not covered" >        this.setState({ books: uniqueBooks });</span>
      }
    });
  };
&nbsp;
  clearBooks = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ books: [] });</span>
  }
&nbsp;
  render() {
    return (
        &lt;div className="search-books"&gt;
        &lt;div className="search-books-bar"&gt;
          &lt;Link className="close-search" to="/"&gt;Close&lt;/Link&gt;
          &lt;div className="search-books-input-wrapper"&gt;
            &lt;SearchBox
              handleSearch={ this.handleSearch }
              onClearSearch={ this.clearBooks }
            /&gt;
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;div className="search-books-results"&gt;
          { this.state.books.length &gt; 0 &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >            &lt;BooksGrid books={ this.state.books } /&gt;</span>
          }
        &lt;/div&gt;
      &lt;/div&gt;
    );
  };
}
&nbsp;
function <span class="fstat-no" title="function not covered" >removeDuplicates(</span>arr, prop) {
  let mappedArr = <span class="cstat-no" title="statement not covered" >arr.map(<span class="fstat-no" title="function not covered" >ma</span>ppedObj =&gt; <span class="cstat-no" title="statement not covered" >mappedObj[prop])</span>;</span>
&nbsp;
  let uniq = <span class="cstat-no" title="statement not covered" >arr.filter(<span class="fstat-no" title="function not covered" >(e</span>lement, index, array) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    return mappedArr.indexOf(element[prop]) === index;</span>
  });
&nbsp;
<span class="cstat-no" title="statement not covered" >  return uniq;</span>
}
&nbsp;
export default SearchBooks;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 11 2017 00:33:58 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
