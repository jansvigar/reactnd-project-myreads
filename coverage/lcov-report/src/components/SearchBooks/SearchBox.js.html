<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/SearchBooks/SearchBox.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/SearchBooks</a> SearchBox.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.78% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.41% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/17</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
&nbsp;
class SearchBox extends Component {
  state = {
    query: ''
  };
&nbsp;
  componentWillMount() {
    this.handleSearchDebounced = debounce(<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >      this.props.handleSearch(this.state.query);</span>
    }, 500);
  };
&nbsp;
  updateQuery = <span class="fstat-no" title="function not covered" >(q</span>uery) =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ query });</span>
  }
&nbsp;
  handleOnChange = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
    let inputValue = <span class="cstat-no" title="statement not covered" >e.target.value;</span>
<span class="cstat-no" title="statement not covered" >    this.props.onClearSearch();</span>
<span class="cstat-no" title="statement not covered" >    this.updateQuery(inputValue);</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if(inputValue) {</span>
<span class="cstat-no" title="statement not covered" >      this.handleSearchDebounced();</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this.handleSearchDebounced.cancel();</span>
    }
  }
&nbsp;
  render() {
    return (
      &lt;input type="text"
          placeholder="Search by title or author"
          value={ this.state.query }
          onChange={ this.handleOnChange }
      /&gt;
    );
  };
}
&nbsp;
SearchBox.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  onClearSearch: PropTypes.func.isRequired
}
&nbsp;
function debounce(callback, wait, context = this) {
  let timeout;
  function <span class="fstat-no" title="function not covered" >debounced(</span>...args) {
<span class="cstat-no" title="statement not covered" >    clearTimeout(timeout);</span>
<span class="cstat-no" title="statement not covered" >    timeout = setTimeout(<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >callback.apply(context, args),</span> wait);</span>
  }
&nbsp;
  debounced.cancel = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    if(timeout !== undefined) {</span>
<span class="cstat-no" title="statement not covered" >      clearTimeout(timeout);</span>
    }
  };
&nbsp;
  return debounced;
}
&nbsp;
export default SearchBox;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 11 2017 00:33:58 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
