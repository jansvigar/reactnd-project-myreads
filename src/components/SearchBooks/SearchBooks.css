/* search page */

.search-books-bar {
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 5;
  display: flex;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 0 6px rgba(0,0,0,0.23);
}
.search-books-input-wrapper {
  flex: 1;
  background: #e9e;
}
.search-books-bar input {
  width: 100%;
  padding: 15px 10px;
  font-size: 1.25em;
  border: none;
  outline: none;
}

.close-search {
  display: block;
  top: 20px;
  left: 15px;
  width: 50px;
  height: 53px;
  background: white;
  background-image: url('../../icons/arrow-back.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: 28px;
  font-size: 0;
}

.search-books-results {
  padding: 80px 10px 20px;

}

/* sort books */

.sort-books-wrapper {
  justify-content: center;
  font-size: 0.9em;
  width: 100%;
}

.sort-books {
  max-width: 340px;
  margin: auto;
  padding: 10px 10px;
  display: flex;
  flex-wrap: wrap;
}

.sort-books label {
  margin: auto 10px;
  flex: 0 1 auto;
}

.sort-books select {
  font-size: inherit;
  width: 70%;
  flex: 1 1 auto;
  padding: 5px 5px;
  border-radius: 5px;
  background: #f9fbff;
}
