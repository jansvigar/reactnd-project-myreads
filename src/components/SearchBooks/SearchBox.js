import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SearchBox extends Component {
  state = {
    query: ''
  };

  componentWillMount() {
    this.handleSearchDebounced = debounce(() => {
      this.props.handleSearch(this.state.query);
    }, 1000);
  };

  updateQuery = (query) => {
    this.setState({ query });
  }

  onChange = (e) => {
    let inputValue = e.target.value;
    this.props.onClearSearch();
    this.updateQuery(inputValue);

    if(inputValue) {
      this.handleSearchDebounced();
    } else {
      this.handleSearchDebounced.cancel();
    }
  }

  render() {
    return (
      <input type="text"
          placeholder="Search by title or author"
          value={ this.state.query }
          onChange={ this.onChange }
      />
    );
  };
}

SearchBox.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  onClearSearch: PropTypes.func.isRequired
}

function debounce(callback, wait, context = this) {
  let timeout;
  function debounced(...args) {
    clearTimeout(timeout);
    timeout = setTimeout(() => callback.apply(context, args), wait);
  }

  debounced.cancel = () => {
    if(timeout !== undefined) {
      clearTimeout(timeout);
    }
  };

  return debounced;
}

export default SearchBox;
