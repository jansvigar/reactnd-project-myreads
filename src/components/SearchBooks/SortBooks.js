import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SortBooks extends Component {
  render() {
    return (
      <div className="sort-books-wrapper">
        <div className="sort-books">
          <label>Sort by:</label>
          <select onChange={ e => this.props.handleSort(e.target.value) }>
            <option value='title_asc'>Title: A to Z</option>
            <option value='title_desc'>Title: Z to A</option>
            <option value='rating_desc'>Average Rating: Highest to Lowest</option>
            <option value='rating_asc'>Average Rating: Lowest to Highest</option>
          </select>
        </div>
      </div>
    );
  };
}

SortBooks.propTypes = {
  handleSort: PropTypes.func.isRequired
};

export default SortBooks;
