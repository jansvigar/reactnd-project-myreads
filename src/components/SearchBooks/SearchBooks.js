import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import * as BooksAPI from '../../BooksAPI';
import BooksGrid from '../BooksGrid/BooksGrid';
import SearchBox from './SearchBox';
import './SearchBooks.css';

class SearchBooks extends Component {
  state = {
    books: []
  };

  handleSearch = (query) => {
    BooksAPI.search(query).then(books => {
      if(!books.error) {
        let uniqueBooks = removeDuplicates(books, 'id');
        this.setState({ books: uniqueBooks });
      }
    });
  };

  clearBooks = () => {
    this.setState({ books: [] });
  }

  render() {
    return (
        <div className="search-books">
        <div className="search-books-bar">
          <Link className="close-search" to="/">Close</Link>
          <div className="search-books-input-wrapper">
            <SearchBox
              handleSearch={ this.handleSearch }
              onClearSearch={ this.clearBooks }
            />
          </div>
        </div>
        <div className="search-books-results">
          { this.state.books.length > 0 &&
            <BooksGrid books={ this.state.books } />
          }
        </div>
      </div>
    );
  };
}

function removeDuplicates(arr, prop) {
  let mappedArr = arr.map(mappedObj => mappedObj[prop]);

  let uniq = arr.filter((element, index, array) => {
    return mappedArr.indexOf(element[prop]) === index;
  });

  return uniq;
}

export default SearchBooks;
