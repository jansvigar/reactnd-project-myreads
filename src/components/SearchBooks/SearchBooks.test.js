import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBooks from './SearchBooks';
import SearchBox from './SearchBox';

const searchBoxProps = {
  handleSearch: jest.fn(),
  onClearSearch: jest.fn()
};

describe('<SearchBooks />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SearchBooks />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('<SearchBox />', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<SearchBox {...searchBoxProps} />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

});
