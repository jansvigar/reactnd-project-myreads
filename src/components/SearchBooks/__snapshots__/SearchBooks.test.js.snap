// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBooks /> <SearchBox /> should render correctly 1`] = `
<input
  onChange={[Function]}
  placeholder="Search by title or author"
  type="text"
  value=""
/>
`;

exports[`<SearchBooks /> should render correctly 1`] = `
<div
  className="search-books"
>
  <div
    className="search-books-bar"
  >
    <Link
      className="close-search"
      replace={false}
      to="/"
    >
      Close
    </Link>
    <div
      className="search-books-input-wrapper"
    >
      <SearchBox
        handleSearch={[Function]}
        onClearSearch={[Function]}
      />
    </div>
  </div>
  <div
    className="search-books-results"
  />
</div>
`;
